### PR TITLE
Bump Grpc.Core dependency to 2.31.0-pre2

### PR DESCRIPTION
**Pull request by jtattermusch**
I'll cut the release branch v2.31.x after this.

### CODE REVIEW

File: build/dependencies.props [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <BenchmarkDotNetPackageVersion>0.12.1</BenchmarkDotNetPackageVersion>
     <GoogleProtobufPackageVersion>3.12.2</GoogleProtobufPackageVersion>
     <GrpcDotNetPackageVersion>2.30.0</GrpcDotNetPackageVersion>  <!-- Used by example projects -->
-    <GrpcPackageVersion>2.30.0</GrpcPackageVersion>
+    <GrpcPackageVersion>2.31.0-pre2</GrpcPackageVersion>
```
Comment (JamesNK) [inline build/dependencies.props:6]: Why pre2? We don't have a pre1.
```suggestion
    <GrpcPackageVersion>2.31.0-pre1</GrpcPackageVersion>
```

Comment (jtattermusch) [reply build/dependencies.props:6]: Grpc.Core pre2 is the latest what we have currently.  (note that this is not version of grpc-dotnet, it's the version of Grpc.Core we import).

Comment (JamesNK) [reply build/dependencies.props:6]: Ok. I guess it makes sense to keep them in sync. grpc-dotnet pre1 depending on Grpc.Core.Api pre2 would be weird.

**Comment by jtattermusch** [comment]
```
/home/runner/work/grpc-dotnet/grpc-dotnet/testassets/InteropTestsClient/InteropTestsClient.csproj : error NU1605: Detected package downgrade: Newtonsoft.Json from 12.0.3 to 12.0.2. Reference the package directly from the project to select a different version.  [/home/runner/work/grpc-dotnet/grpc-dotnet/Grpc.DotNet.sln]
/home/runner/work/grpc-dotnet/grpc-dotnet/testassets/InteropTestsClient/InteropTestsClient.csproj : error NU1605:  InteropTestsClient -> Grpc.Auth 2.31.0-pre2 -> Google.Apis.Auth 1.46.0 -> Google.Apis.Core 1.46.0 -> Newtonsoft.Json (>= 12.0.3)  [/home/runner/work/grpc-dotnet/grpc-dotnet/Grpc.DotNet.sln]
/home/runner/work/grpc-dotnet/grpc-dotnet/testassets/InteropTestsClient/InteropTestsClient.csproj : error NU1605:  InteropTestsClient -> Newtonsoft.Json (>= 12.0.2) [/home/runner/work/grpc-dotnet/grpc-dotnet/Grpc.DotNet.sln]
```

**Comment by JunTaoLuo** [comment]
I pushed a fix. Essentially it was complaining that we were explicitly specifying NewtonsoftJsonPackageVersion to 12.0.2 which conflicted with what Grpc.Auth was using.

**Pull request merged**